src/ctapipe/irf/preprocessing.py

```diff
@@ -85,15 +79,16 @@ class EventPreprocessor(Component):
     fixed_columns = List(
         Unicode(),
         default_value=["obs_id", "event_id", "true_energy", "true_az", "true_alt"],
-        help="Columns to always keep from the original input",
+        help="Columns to keep from the input table without renaming.",
     ).tag(config=True)
 
-    # Optional user override
-    columns_to_rename_override = Dict(
-        key_trait=Unicode(),
-        value_trait=Unicode(),
-        default_value={},
-        help="Override of columns to rename. If empty, they are generated dynamically.",
+    columns_to_rename_override = Any(
```

Comment:
Why did you change to Any from the correct `Dict[str, str]`?

Reply:
If the problem was the `None`, the correct way is to specify `allow_none=True`.

Reply:
I also do not understand why we need the None case.

Can't we keep the default but allow to set to an empty dict to disable?

For what do we need the `None`?

Reply:
Ah, I see, the default depends on other traits.

Traitlets has a mechanism for this. Define a function and decorate it with `@default("columns_to_rename")`.